### ISSUE TITLE
Python - aggregate function issues

### ISSUE THREAD

**Issue by Nitrooo** [ccxt-dev/ccxt#309]
1) Bitfinex v2 API returns book items in [price, volume, timestamp] format, while aggregate function expects [price, volume] format.

`order_book = ccxt.bitfinex2().fetch_l2_order_book('BTC/USD')`

> Traceback (most recent call last):
> ...
>     order_book = ccxt.bitfinex2().fetch_l2_order_book('BTC/USD')
>   File "D:\Python\Python36-32\lib\site-packages\ccxt\exchange.py", line 787, in fetch_l2_order_book
>     'bids': self.sort_by(self.aggregate(orderbook['bids']), 0, True),
>   File "D:\Python\Python36-32\lib\site-packages\ccxt\exchange.py", line 511, in aggregate
>     for [price, volume] in bidasks:
> ValueError: too many values to unpack (expected 2)

---

2) ordered.iteritems() should be changed to ordered.items(), as iteritems() has been removed in Python 3.

`order_book = ccxt.cex().fetch_l2_order_book('BTC/USD')`

> Traceback (most recent call last):
> ...
>     order_book = ccxt.cex().fetch_l2_order_book('BTC/USD')
>   File "D:\Python\Python36-32\lib\site-packages\ccxt\exchange.py", line 787, in fetch_l2_order_book
>     'bids': self.sort_by(self.aggregate(orderbook['bids']), 0, True),
>   File "D:\Python\Python36-32\lib\site-packages\ccxt\exchange.py", line 514, in aggregate
>     for price, volume in ordered.iteritems():
> AttributeError: 'collections.OrderedDict' object has no attribute 'iteritems'

**Comment by kroitor** [comment]
Hi, @Nitrooo ! 

Thx for opening this issue!

> Bitfinex v2 API returns book items in [price, volume, timestamp] format, while aggregate function expects [price, volume] format.

I fixed the bitfinex2 order book, should work normally now. 

> ordered.iteritems() should be changed to ordered.items(), as iteritems() has been removed in Python 3

We support both Python 2 and Python 3 – so, my question is: is there a way to write that line so that I could execute it seamlessly in both 2 and 3 ?

Thanks again!

**Comment by Nitrooo** [comment]
> We support both Python 2 and Python 3 – so, my question is: is there a way to write that line so that I could execute it seamlessly in both 2 and 3 ?

I'm no Python expert at all, maybe this link could help: https://stackoverflow.com/questions/10458437/what-is-the-difference-between-dict-items-and-dict-iteritems
Maybe `dict.viewitems()` is what we're looking for.

**Comment by kroitor** [comment]
Ok, I fixed both of these issues, please update to v.1.9.108+ and try it, should work normally now.  I'm closing it for now, if you don't mind, feel free to reopen it or ask your questions, if any. Thx for your feedback!

**Comment by Nitrooo** [comment]
I found an official guide about porting dictionary iteration from Python 2 to 3 and common subset.
http://legacy.python.org/dev/peps/pep-0469/
As noted on stackoverflow:
> The recommended approach appears to be to use the `iteritems` function of the `future.utils` or `six` modules instead.

**Comment by kroitor** [comment]
@Nitrooo there's a better way without six:

```
items = list(dict.items())
for key, value in items:
    print(key, value)
```

**Comment by kroitor** [comment]
More to it: http://python-future.org/compatible_idioms.html

**Comment by Nitrooo** [comment]
> More to it: http://python-future.org/compatible_idioms.html

```
# Python 2 and 3: option 1
itemlist = list(heights.items())    # inefficient on Py2
```

But that soultion is marked as inefficient! 😃

**Comment by kroitor** [comment]
@Nitrooo that is not critical in our case, because:

1. efficiency matters, if you are going to aggregate tons of order books every second, which isn't the case for most of trading applications
2. this is only done once upon fetching the orderbook
3. you can't possibly fetch so many orderbooks in a limited amount of time so that the efficiency of converting 'items()' to a list becomes critical
4. Python 2 is going to die soon

**Comment by Nitrooo** [comment]
fetch_order_book() doesn't work anymore with Bitfinex v2 on ccxt v.1.9.107+. It works fine on v.1.9.106.
I have not much time to investigate right now, but you get the timestamp in place of the price, for example.

**Comment by kroitor** [comment]
Oh, sorry, my bad, will fix it right away

**Comment by kroitor** [comment]
Ok, now it's fixed (1.9.110+). Thx!

**Comment by Nitrooo** [comment]
Thank you!
